src/sagemaker/payloads.py

```diff
@@ -0,0 +1,116 @@
+) -> Optional[List[str]]:
+    """Retrieves the compatible payloads for the model matching the given arguments.
+
+    Args:
+        region (str): The AWS Region for which to retrieve the Jumpstart model payloads.
+        model_id (str): The model ID of the JumpStart model for which to retrieve
+            the model payloads.
+        model_version (str): The version of the JumpStart model for which to retrieve
+            the model payloads.
+        serialize (bool): Whether to serialize byte-stream valued payloads by downloading
+            binary files from s3 and applying encoding, or to keep payload in pre-serialized
+            state. Set this option to False if you want to avoid s3 downloads or if you
+            want to inspect the payload in a human-readable form. (Default: True).
+        tolerate_vulnerable_model (bool): ``True`` if vulnerable versions of model
+            specifications should be tolerated without raising an exception. If ``False``, raises an
+            exception if the script used by this version of the model has dependencies with known
+            security vulnerabilities. (Default: False).
+        tolerate_deprecated_model (bool): ``True`` if deprecated versions of model
+            specifications should be tolerated without raising an exception. If ``False``, raises
+            an exception if the version of the model is deprecated. (Default: False).
+        sagemaker_session (sagemaker.session.Session): A SageMaker Session
+            object, used for SageMaker interactions. If not
+            specified, one is created using the default AWS configuration
+            chain. (Default: sagemaker.jumpstart.constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION).
+    Returns:
+        str: The model artifact S3 URI for the corresponding model.
+
+    Raises:
+        NotImplementedError: If the scope is not supported.
+        ValueError: If the combination of arguments specified is not supported.
+        VulnerableJumpStartModelError: If any of the dependencies required by the script have
+            known security vulnerabilities.
+        DeprecatedJumpStartModelError: If the version of the model is deprecated.
+    """
+    if not jumpstart_utils.is_jumpstart_model_input(model_id, model_version):
+        raise ValueError(
+            "Must specify JumpStart `model_id` and `model_version` when retrieving payloads."
+        )
+
+    unserialized_payload_dict: Optional[
+        Dict[str, JumpStartSerializablePayload]
+    ] = artifacts._retrieve_default_payloads(
+        model_id,
+        model_version,  # type: ignore
+        region,
+        tolerate_vulnerable_model,
+        tolerate_deprecated_model,
+        sagemaker_session=sagemaker_session,
+    )
+
+    if unserialized_payload_dict is None:
+        return None
+
+    unserialized_payloads: List[JumpStartSerializablePayload] = list(
+        unserialized_payload_dict.values()
+    )
```

Comment:
here aren't we buffering a list of content-embedded payloads in memory?

Let's sync up offline on this point please.